### PR TITLE
applications: nrf_desktop: Update HID forward documentation

### PR DESCRIPTION
**Pull request by MarekPieta**
Change documents CONFIG_DESKTOP_HID_FORWARD_SUBSCRIBER_COUNT Kconfig option.

Jira: NCSDK-30002

### CODE REVIEW

**Review by kapi-no — APPROVED** [review]

File: applications/nrf_desktop/doc/hid_forward.rst [hunk 1]
```diff
@@ -50,6 +50,10 @@ Complete the following steps to configure the module:
    Those options affect :ref:`nrf_desktop_usb_state` that subscribes for HID boot reports.
    The Dongle forwards HID reports from both mouse and keyboard, and so either option works if you want to have the Dongle work as boot mouse or boot keyboard.
    For more information about the configuration of the HID boot protocol, see the boot protocol configuration section in the :ref:`nrf_desktop_usb_state` documentation.
+#. nRF Desktop dongle can use one or more instances of the USB HID class.
```
Comment (peknis) [inline applications/nrf_desktop/doc/hid_forward.rst:53]: The whole sequence of steps here does not really seem to me as clear steps. The steps should be chrystal clear, always starting with an imperative word. After that, you can always spice up the step with additional and necessary information.

**Review by peknis — CHANGES_REQUESTED** [review]

File: applications/nrf_desktop/doc/hid_forward.rst (hunk 1 above)
Comment (MarekPieta) [reply applications/nrf_desktop/doc/hid_forward.rst:53]: You are right, but some steps are optional/conditional and might require introducing additional context before informing about the needed configuration update. Still, I think it's good to have them on the list to ensure users would not overlook the important configuration checks (we already follow this approach in multiple modules' docs). We also try to highlight some of the important configuration changes that are automatically applied under the hood.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -202,7 +202,8 @@ nRF5340 Audio
 nRF Desktop
 -----------
 
-|no_changes_yet_note|
+* Added a Kconfig option that configures number of HID subscribers in :ref:`nrf_desktop_hid_forward` (:ref:`CONFIG_DESKTOP_HID_FORWARD_SUBSCRIBER_COUNT <config_desktop_app_options>`).
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:205]:
```suggestion
* Added a Kconfig option that configures the number of HID subscribers in :ref:`nrf_desktop_hid_forward` (:ref:`CONFIG_DESKTOP_HID_FORWARD_SUBSCRIBER_COUNT <config_desktop_app_options>`).
```

**Review by peknis — APPROVED** [review]
Approved with a nit. Further improvements later as agreed offline.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -202,7 +202,8 @@ nRF5340 Audio
 nRF Desktop
 -----------
 
-|no_changes_yet_note|
+* Added a Kconfig option that configures the number of HID subscribers in :ref:`nrf_desktop_hid_forward` (:ref:`CONFIG_DESKTOP_HID_FORWARD_SUBSCRIBER_COUNT <config_desktop_app_options>`).
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:205]: Remove the changelog entry from this PR and add it as a comment to Divya's https://github.com/nrfconnect/sdk-nrf/pull/18192.

**Review by zycz — APPROVED** [review]

**Pull request merged**